nifi-commons/nifi-repository-encryption/src/main/java/org/apache/nifi/repository/encryption/configuration/kms/StandardRepositoryKeyProviderFactory.java

```diff
@@ -0,0 +1,216 @@
+                final String message = String.format("Key Provider Property [%s] not configured", classProperty);
+                throw new EncryptedConfigurationException(message);
+            } else {
+                encryptionKeyProvider = getEncryptionKeyProvider(implementationClass);
+            }
+        } else {
+            for (final EncryptionKeyProvider currentEncryptKeyProvider : EncryptionKeyProvider.values()) {
+                if (currentEncryptKeyProvider.toString().equals(sharedKeyProvider)) {
+                    encryptionKeyProvider = currentEncryptKeyProvider;
+                }
+            }
+        }
+
+        if (encryptionKeyProvider == null) {
+            final String message = String.format("Key Provider [%s] not found for Repository Type [%s] ", sharedKeyProvider, encryptedRepositoryType);
+            throw new EncryptedConfigurationException(message);
+        }
+
+        return encryptionKeyProvider;
+    }
+
+    private EncryptionKeyProvider getEncryptionKeyProvider(final String implementationClass) {
+        EncryptionKeyProvider encryptionKeyProvider;
+
+        if (implementationClass.endsWith(FileBasedKeyProvider.class.getSimpleName())) {
+            encryptionKeyProvider = EncryptionKeyProvider.FILE_PROPERTIES;
+        } else if (implementationClass.endsWith(KeyStoreKeyProvider.class.getSimpleName())) {
+            encryptionKeyProvider = EncryptionKeyProvider.KEYSTORE;
+        } else if (implementationClass.endsWith(StaticKeyProvider.class.getSimpleName())) {
+            encryptionKeyProvider = EncryptionKeyProvider.NIFI_PROPERTIES;
+        } else {
+            final String message = String.format("Key Provider Class [%s] not supported", implementationClass);
+            throw new IllegalArgumentException(message);
+        }
+
+        return encryptionKeyProvider;
+    }
+
+    private Map<String, String> getEncryptionKeys(final EncryptedRepositoryType encryptedRepositoryType, final NiFiProperties niFiProperties) {
+        switch (encryptedRepositoryType) {
+            case CONTENT:
+                return niFiProperties.getContentRepositoryEncryptionKeys();
+            case FLOW_FILE:
+                return niFiProperties.getFlowFileRepoEncryptionKeys();
+            case PROVENANCE:
+                return niFiProperties.getProvenanceRepoEncryptionKeys();
+            default:
+                throw new IllegalArgumentException(String.format("Repository Type [%s] not supported", encryptedRepositoryType));
+        }
+    }
+
+    private KeyProviderConfiguration<?> getKeyProviderConfiguration(final EncryptedRepositoryType encryptedRepositoryType,
+                                                                    final EncryptionKeyProvider encryptionKeyProvider,
+                                                                    final NiFiProperties niFiProperties) {
+        if (EncryptionKeyProvider.NIFI_PROPERTIES == encryptionKeyProvider) {
```

Comment:
Part of the purpose of this factory class is to maintain separation between the `NiFiProperties` and the `KeyProvider` implementation classes. There are probably some options to streamline the approach while maintaining the separation, so I will evaluate some options.  The approach of storing repository encryption keys in `NiFiProperties` is not optimal for several reasons, but it could be more straightforward.

Reply:
After further evaluation, `NiFiProperties` has a method named `getRepositoryEncryptionKeys()` that takes the repository type as a parameter, so that simplifies part of the configuration process.